applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -408,16 +424,62 @@ def _create_linear_solver(self):
         return linear_solver
 
     def _create_builder_and_solver(self):
+        if self.settings["block_builder"].GetBool() and self.settings["multi_point_constraints_used"].GetBool():
```

Comment:
"multi_point_constraints_used"  is a flag. 
Also "ResidualBasedBlockBuilderAndSolverWithMpc" is also a block builder and solver.
I think the following should be the logic

if (block_builder){
   if (multi_point_constraints_used)
      make block builder and solver with MPC 
   else 
      make normal block builder and solver
} else {
     make elimination builder and solver 
}

Reply:
Fixed